### PR TITLE
Fix:  hangs if  is empty and SDK NoOp

### LINKED ISSUE #908: [dio] dio.addSentry hangs if dsn is an empty string
_Platform:_
- [ ]  Dart
- [x]  Flutter Android or iOS
- [ ]  Flutter Web

_IDE:_
- [x]  VSCode
- [ ]  IntelliJ/AS
- [ ]  XCode
- [ ]  Other, which one?

_split-debug-info and obfuscate (Flutter Android or iOS) or CanvasKit (Flutter Web):_
- [ ]  Enabled
- [x]  Disabled

_Platform installed with:_
- [x] pub.dev
- [ ] GitHub

Output of the command `flutter doctor -v` below:

```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.0.3, on Arch Linux 5.18.7-arch1-1, locale en_US.UTF-8)
[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
[✓] Chrome - develop for the web
[✓] Linux toolchain - develop for Linux desktop
[✓] Android Studio (version 2021.2)
[✓] IntelliJ IDEA Ultimate Edition (version 2021.2)
[✓] Connected device (3 available)
[✓] HTTP Host Availability

• No issues found!
```

The version of the SDK (See pubspec.lock):
**6.6.1**

---
I have the following issue:

When setting `options.dsn = ""` in debug mode, the call `dio.addSentry()` causes the app to hang indefinitely. 

**Steps to reproduce:**

```dart
await SentryFlutter.init((options) {
    options.dsn = "";
});
dio.addSentry();
```

**Actual result:**
The whole app hangs until hard restarted.

**Expected result:**
The `dio.addSentry()` call succeeds and events are not sent because of the empty dsn.

### PR DESCRIPTION
**Pull request by marandaneto**
## :scroll: Description
Fix: hangs if is empty and SDK NoOp

## :bulb: Motivation and Context
Fixes https://github.com/getsentry/sentry-dart/issues/908
`sdk` is a late var and was not init for the NoOp Hub.

## :pencil: Checklist

- [X] I reviewed submitted code
- [X] I added tests to verify changes
- [ ] I updated the docs if needed
- [X] All tests passing
- [X] No breaking changes

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -13,7 +15,7 @@
 
 ### Fixes
 
-- Send DidBecomeActiveNotification when OOM enabled (#905)
+* Send DidBecomeActiveNotification when OOM enabled (#905)
```
Comment (ueman) [inline CHANGELOG.md:20]: The auto-updater uses `-`, that should probably be aligned?

Comment (marandaneto) [reply CHANGELOG.md:20]: Most likely, good catch, auto formatter changed automatically for me.

Comment (marandaneto) [reply CHANGELOG.md:20]: https://github.com/getsentry/sentry-dart/issues/922

**Review by brustolin — APPROVED** [review]
LGTM

**Pull request merged**